### DISCUSSION TITLE
swc plugin in amplify / lambda hosting

### DISCUSSION THREAD

**Discussion by azgorov** [swc-project/plugins#313] [Q&A]
I am not sure that this is correct place for this question but here it is;

I have very simple swc visitor plugin.

Using amazon linux 23 wsl image testing transform / transformSync functions i got valid result.
The same transform code published in aws amplify. which is linux x64 lambda does not work. It does not give any error - plugin is just ignored.

File path is correct. File seems fine and transform looks ok.

I have added println in first line of "process_transform function" function and while it work on local docker it does not on lambda.

Does anyone had similar issues? Any ideas what can it be and why deployed code is ignored?

 I know that is very special case...

Best
Kalin

**Comment by kwonoj** [comment]
There are certain platform/target does not support plugins. What's the target triple?

**Comment by azgorov** [comment]
Local: x86_64-amazon-linux-gnu
Remote: x86_64-koji-linux-gnu

**Reply by kwonoj** [reply]
x64 gnu is definitely supported afaik. Can you confirm what swc target triple binary is actually loaded? I.e https://www.npmjs.com/package/@swc/core-linux-x64-gnu

**Reply by azgorov** [reply]
I will check but if it is missing node files it will not work at all right? transform is working just not plugin.

**Reply by kwonoj** [reply]
What I meant is what actual .node binary is being loaded in @swc/core. That tells the exact target triple swc determimes.

**Reply by azgorov** [reply]
It seems that hosting removes big files so although it is part of deployment. This results that swc.linux-x64-gnu.node is removed as it is almost 50mb.

I am not sure why transform and parse are working at all in this case but it seems that file is missing.

Thank you for your time. It looks that is is hosting issue.

Thanks
Kalin
